enhancements/migration/cam-non-admin-design.md

```diff
@@ -0,0 +1,351 @@
+      validate Bob is using a secretRef that was granted to him by a
+      cluster-admin without the ability to actually read the secret. This also
+      prevents another user Alice from simply being able to guess Bob’s
+      secretRef.
+   1. https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/
+1. The cluster-administrator then creates a MigCluster resource in
+   `openshift-migration` which includes the usual spec fields:
+   1. Mig SA Secret Ref
+   1. Cluster coordinates
+   ![figure1](./cam-non-admin-1.png)
+#####   Figure 1: Installation of CAM Operator and cluster-admin Setup
+   ![figure2](./cam-non-admin-2.png)
+#####   Figure 2: Cluster Admin creates MigCluster Resource
+6. Bob creates a MigrationSandbox resource which will trigger the operator to
+   provision a new migration controller in openshift-migration namespace that
+   watches only Bob’s namespace (where the MigrationSandbox resource was
+   created). It also provisions and configures a Migration UI instance in Bob’s
+   namespace.
+7. Bob creates secret(s) containing identity tokens for the source (and
+   optionally the target) cluster in bob-cam-namespace.
+   ![figure3](./cam-non-admin-3.png)
+#####   Figure 3: Bob installs CAM into a namespace He is an Admin of
+8. Bob then interfaces with the Migration UI to create a MigPlan resource with
+   the following:
+   1. secretRef to Mig SA token Bob has been granted “USE”
+   1. user identity token secretRef
+   1. List of namespaces Bob wishes to migrate
+9. MigPlan is validated by checking that Bob has been granted “USE” on the
+   given secret and runs through the permissions check using the discovery
+   service to get the list of valid projects for Bob to migrate using Bob’s
+   identity token.
+   1. The migplan controller validates that the list of namespaces is a subset
+      of the namespaces validated with the migcluster.
+   ![figure4](./cam-non-admin-4.png)
+#####   Figure 4: Bob creates a valid Mig Plan
+   ![figure5](./cam-non-admin-5.png)
+#####   Figure 5: Bob creates an invalid Mig Plan
+10. At this point, Bob has a “Ready” migcluster, a “Ready” migplan, and now can
+   create a migmigration. The migmigration needs to perform the validation
+   check one more time prior to running the migration.
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
```

Comment:
There has to a migration strategy for a user who has installed CAM today, that is fully cluster scoped when we upgrade them to operator that is capable of non-admin. Is there some sort of config to manage how the CAM operator will handle this transition?

Reply:
Added some notes about how we are handling the upgrade scenario